src/main/webapp/app/shared/feature-toggle/feature-toggle-hide.directive.ts

```diff
@@ -6,7 +6,7 @@ import { Subscription } from 'rxjs/internal/Subscription';
     selector: '[jhiFeatureToggleHide]',
 })
 export class FeatureToggleHideDirective implements OnInit, OnDestroy {
-    @Input('jhiFeatureToggleHide') feature: FeatureToggle;
+    @Input('jhiFeatureToggleHide') feature: FeatureToggle | undefined;
```

Comment:
Why would we have a `FeatureToggleHideDirective` where the toggle is undefined?

Reply:
It's because of my generic creation of the Navigation Items. Unfortunately, it's not possible to dynamically apply directives. The directive checks for undefined and is aware that it shouldn't do anything for undefined values.